pkg/kconfig/parser.go

```diff
@@ -142,7 +142,7 @@ func (p *parser) QuotedString() string {
 		if ch == '\\' {
 			ch = p.char()
 			switch ch {
-			case '\'', '"', '\\':
+			case '\'', '"', '\\', 'n':
```

Comment:
Why do we need this change?

Reply:
See commit message:
```
linux-next contains "Kconfig: Add option for asm goto w/ tied outputs to
workaround clang-13 bug", which added this elaborate config variable to
init/Kconfig:

 | config CC_HAS_ASM_GOTO_TIED_OUTPUT
 | 	depends on CC_HAS_ASM_GOTO_OUTPUT
 | 	# Detect buggy gcc and clang, fixed in gcc-11 clang-14.
 | 	def_bool $(success,echo 'int foo(int *x) { asm goto (".long (%l[bar]) - .\n": "+m"(*x) ::: bar); return *x; bar: return 0; }' | $CC -x c - -c -o /dev/null)

The current validation in parser.go will complain with "bad quoted
character", because of the '\n' in the quoted string.

Fix it by allowing '\n'.
```